File: LeetCode/Simulation/JS/Leetcode1763-最长的美好字符串.js
var longestNiceSubstring = function(s) {
    n = s.length;
    let start = 0,len = 0;
    for(let l = 0;l < n;l++) {
        let a = 0,b = 0;
        for(let r = l;r < n;r++) {
            if(s[r] >= 'a' && s[r] <= 'z') a |= 1 << (s[r].codePointAt(0) - 'a'.codePointAt(0));
            else b |= 1 << (s[r].codePointAt(0) - 'A'.codePointAt(0));
            if(a == b && r - l + 1 > len) {
                len = r - l + 1;
                start = l;
            }
        }
    }
    return s.substr(start,len);
};

File: LeetCode/Simulation/Golang/Leetcode1763-最长的美好字符串.go
package sbgo 

func longestNiceSubstring(s string) string {
    start,length,n := 0,0,len(s)
    for l := 0;l < n;l++ {
        a,b := 0,0
        for r := l;r < n;r++ {
            if s[r] >= 'a' && s[r] <= 'z'{
                a |= 1 << (s[r] - 'a')
            }else {
                b |= 1 << (s[r] - 'A')
            }
            if a == b && r - l + 1 > length {
                length = r - l + 1
                start = l
            }
        }
    }
    return s[start:start + length]
}